facebookresearch/metaseq
# Add Aim logging to progress_bar

**Patch Description**
- Added `AimProgressBarWrapper` class; an implementation of `BaseProgressBar` using Aim, an open-source experiment tracker: https://github.com/aimhubio/aim.
- Added two new arguments to `CommonConfig` used to control `AimProgressBarWrapper` creation/usage:
  - `aim_repo`: path to the Aim repository.
  - `aim_run_hash`: If specified, will be used to determine `aim.Run` for metadata tracking. If skipped, `checkpoint.save_dir` will be used to find and re-open the existing `Run` or if no matching `Run` is found, new one is created.

**Testing steps**
Make sure `progress_bar` logs data to selected outputs, with Aim logger enabled/disabled.
Run unit-tests.

- [ ] Was this discussed/approved via a Github issue?
- [x] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
